src/lib.rs

```diff
@@ -623,16 +636,17 @@ impl<'a> ProgramGroup<'a> {
     ///     ProgramBlueprint::new(&std::fs::read(program).expect("Could not open file"), None)
     ///         .expect("Could not open test object file");
     ///
-    /// ProgramGroup::new(None);
+    /// ProgramGroup::new(None, 32);
     /// ```
-    pub fn new(event_buffer_size: Option<usize>) -> ProgramGroup<'a> {
-        let event_buffer_size = event_buffer_size.unwrap_or(1024);
+    pub fn new(buffer_capacity: Option<usize>, element_size: usize) -> ProgramGroup<'a> {
```

Comment:
Since we're breaking the interface anyway, I'd like to refactor these as builder options with sensible defaults. Loading and reading a perfmap isn't strictly required, and there won't always be a crossbeam channel returned, so these are technically optional. It's up to the user to know whether or not they need the perfmap and provide the element size.